File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/BlackjackGame.py
import random


class BlackjackGame:
    def __init__(self):
        self.deck = self.create_deck()
        self.player_hand = []
        self.dealer_hand = []

    def create_deck(self):
        suits = ["S", "C", "D", "H"]
        ranks = ["A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"]
        deck = [rank + suit for suit in suits for rank in ranks]
        random.shuffle(deck)
        return deck

    def calculate_hand_value(self, hand):
        value = 0
        num_aces = 0
        for card in hand:
            rank = card[:-1]  # all characters except last (suit)
            if rank.isdigit():
                value += int(rank)
            elif rank in ("J", "Q", "K"):
                value += 10
            elif rank == "A":
                value += 11
                num_aces += 1
        while value > 21 and num_aces > 0:
            value -= 10
            num_aces -= 1
        return value

    def check_winner(self, player_hand, dealer_hand):
        player_value = self.calculate_hand_value(player_hand)
        dealer_value = self.calculate_hand_value(dealer_hand)
        if player_value > 21 and dealer_value > 21:
            return "Player wins" if player_value <= dealer_value else "Dealer wins"
        elif player_value > 21:
            return "Dealer wins"
        elif dealer_value > 21:
            return "Player wins"
        else:
            return "Player wins" if player_value > dealer_value else "Dealer wins"


if __name__ == "__main__":
    game = BlackjackGame()
    print(game.deck)
    print(game.calculate_hand_value(["QD", "9D", "JC", "QH", "AS"]))
    print(game.check_winner(["QD", "9D", "JC", "QH", "AS"], ["QD", "9D", "JC", "QH", "2S"]))

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_cpp/code_output/BlackjackGame.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <random>
#include <cctype>

class BlackjackGame {
public:
    std::vector<std::string> deck;
    std::vector<std::string> playerHand;
    std::vector<std::string> dealerHand;

    BlackjackGame() {
        deck = createDeck();
    }

    std::vector<std::string> createDeck() {
        std::vector<std::string> deck;
        const std::string suits[] = {"S", "C", "D", "H"};
        const std::string ranks[] = {"A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"};
        
        for (const auto& suit : suits) {
            for (const auto& rank : ranks) {
                deck.push_back(rank + suit);
            }
        }
        
        std::random_device rd;
        std::mt19937 g(rd());
        std::shuffle(deck.begin(), deck.end(), g);
        
        return deck;
    }

    int calculateHandValue(const std::vector<std::string>& hand) {
        int value = 0;
        int numAces = 0;
        
        for (const auto& card : hand) {
            std::string rank = card.substr(0, card.length() - 1);
            
            // Equivalent to Java's rank.matches("\\d+")
            bool isDigit = !rank.empty() && std::all_of(rank.begin(), rank.end(), 
                [](unsigned char c) { return std::isdigit(c); });
                
            if (isDigit) {
                value += std::stoi(rank);
            } else if (rank == "J" || rank == "Q" || rank == "K") {
                value += 10;
            } else if (rank == "A") {
                value += 11;
                numAces += 1;
            }
        }
        
        while (value > 21 && numAces > 0) {
            value -= 10;
            numAces -= 1;
        }
        
        return value;
    }

    std::string checkWinner(const std::vector<std::string>& playerHand, const std::vector<std::string>& dealerHand) {
        int playerValue = calculateHandValue(playerHand);
        int dealerValue = calculateHandValue(dealerHand);
        
        if (playerValue > 21 && dealerValue > 21) {
            return playerValue <= dealerValue ? "Player wins" : "Dealer wins";
        } else if (playerValue > 21) {
            return "Dealer wins";
        } else if (dealerValue > 21) {
            return "Player wins";
        } else {
            return playerValue > dealerValue ? "Player wins" : "Dealer wins";
        }
    }
};

// Helper function to mimic Java's ArrayList.toString() output format
std::string listToString(const std::vector<std::string>& list) {
    std::string result = "[";
    for (size_t i = 0; i < list.size(); ++i) {
        if (i > 0) {
            result += ", ";
        }
        result += list[i];
    }
    result += "]";
    return result;
}

int main() {
    BlackjackGame game;
    
    std::cout << listToString(game.deck) << std::endl;
    std::cout << game.calculateHandValue({"QD", "9D", "JC", "QH", "AS"}) << std::endl;
    std::cout << game.checkWinner({"QD", "9D", "JC", "QH", "AS"}, {"QD", "9D", "JC", "QH", "2S"}) << std::endl;
    
    return 0;
}